server/db/model/Announcement.js

```diff
@@ -61,6 +69,15 @@ schema.query.beforeNow = function() {
     });
 };
 
+// Allow us to query for announcements after a date
+schema.query.since = function(since) {
+    return this.find({
+        broadcastTime: {
+            $gte: new Date(parseInt(since || 0))
```

Comment:
If the client is updated to timestamp 10, announcement X was broadcast at timestamp 5, and announcement X is deleted at timestamp 15, when the client asks for since=10, I don't think this is going to give client back that X was deleted no? You may need to have a lastUpdated field on each announcement and use that

Reply:
@connorkrupp there is a new updatedAt field that should be attached to all documents once this PR is merged in (thanks to defaultSchema/Options). Mongoose will automagically update those fields when any change to the document occurs. Will that work?

Reply:
Yeah thats even better than great